src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -978,10 +978,6 @@ bool PSParallelCompact::invoke(bool clear_all_soft_refs) {
   SvcGCMarker sgcm(SvcGCMarker::FULL);
   IsSTWGCActiveMark mark;
 
-  ParallelScavengeHeap* heap = ParallelScavengeHeap::heap();
-  clear_all_soft_refs = clear_all_soft_refs
-                     || heap->soft_ref_policy()->should_clear_all_soft_refs();
-
```

Comment:
* the comment for this method needs to be fixed
* the CR description in JIRA should be clarified. This change does not remove all occurrences of `_soft_ref_policy` in Parallel.
* the parameter name `clear_all_soft_refs` in the declaration and the definition are different

Reply:
>     * the CR description in JIRA should be clarified. This change does not remove all occurrences of `_soft_ref_policy` in Parallel.

It does. `_soft_ref_policy` should probably be moved to the collectors actually using it (before cleaning up its use, I think similar changes can be done for G1 and Serial)